### ISSUE TITLE
Bugs in Learning Rate Scheduler?

### ISSUE THREAD

**Issue by TiankaiHang** [Jia-Research-Lab/Context-Aware-Consistency#10]
![image](https://user-images.githubusercontent.com/52823230/119956741-fdae5f00-bfd3-11eb-97a6-c952c329f604.png)

May the learning rate used not 'poly'?

`def get_lr(self):
    T = self.last_epoch * self.iters_per_epoch + self.cur_iter
    factor =  pow((1 - 1.0 * T / self.N), 0.9)
    if self.warmup_iters > 0 and T < self.warmup_iters:
        factor = 1.0 * T / self.warmup_iters
    self.cur_iter %= self.iters_per_epoch
    self.cur_iter += 1
    assert factor >= 0, 'error in lr_scheduler'
    return [base_lr * factor for base_lr in self.base_lrs]`

However, the lr updated as follows,
`self.lr_scheduler.step(epoch=epoch-1)`

Is that a bug or a specific design?
https://github.com/pytorch/pytorch/blob/41054f2ab5bb39d28a3eb8497f1a65b42385a996/torch/optim/lr_scheduler.py#L155

**Comment by X-Lai** [comment]
Sorry, I may not get your point.
I think the learning rate scheduler is correct. The poly scheduler should be like this figure, because the exponent is set to 0.9, so the decay curve may look like a line.

**Comment by X-Lai** [comment]
I will close this issue. If you have further question, welcome to reopen it.
